lib/mapObjects/ObjectTemplate.cpp

```diff
@@ -231,21 +224,14 @@ void ObjectTemplate::readMap(CBinaryReader & reader)
 
 	reader.readUInt16();
 	ui16 terrMask = reader.readUInt16();
-	for(size_t i = Terrain::FIRST_REGULAR_TERRAIN; i < Terrain::ROCK; i++)
+	for(TerrainId i = ETerrainId::FIRST_REGULAR_TERRAIN; i < ETerrainId::ORIGINAL_TERRAIN_COUNT; ++i)
```

Comment:
ORIGINAL_TERRAIN_COUNT and ROCK are the same?

Reply:
No, however that bit will be always at 0.
That being said, I'll recheck usages of ROCK & ORIGINAL_TERRAIN_COUNT.
Code is correct, but I find it weird to use ROCK (or any other identifier) as loop end point